### ISSUE TITLE
Injection of batch artifact properties does not work

### ISSUE THREAD

**Issue by chschroe74** [jberet/jsr352#273]
According to https://jberet.gitbooks.io/jberet-user-guide/content/batch_properties/, I should be able to inject batch artifact properties using the @BatchProperty annotation. This mechanism is even used in several of the existing test cases (e.g. "SleepBatchletTest"); however, when I set a breakpoint in the "process" method of that batchlet, the "sleepMinutes" field is always 0.

I could not find any test case specifically covering batch artifact properties. (The "InjectBatchletTest" only validates job and step properties.) Hence, my assumption is that it is a bug that should be fixed.

**Comment by chschroe74** [comment]
Indeed, in the existing test "SleepBatchletTest", the artifact property refers to a job parameter, which is only set for one of the tests. However, when I change the corresponding batch definition "org.jberet.se.test.sleepBatchlet.xml" like this ...

```
<batchlet ref="sleepBatchlet">
    <properties>
        <property name="sleep.minutes" value="5"></property>
    </properties>
</batchlet>
```

... and add the following to the "SleepBatchlet.process" method ...

```
@Override
public String process() throws Exception {
    System.out.println("sleepMinutes = " + sleepMinutes);
    processThread = Thread.currentThread();
    ...
``` 

... the output is always: "sleepMinutes = 0"

Is it not allowed to specify an artifact property as a constant value instead of a job parameter reference? I could not find anything in the specs.

**Comment by chengfang** [comment]
constant value is perfectly fine, and should work. Not sure why it's not working, but will look into it.

**Comment by chengfang** [comment]
with a constant value, I can see it printed in process method:

`mvn test -Dtest=org.jberet.se.test.SleepBatchletTest#sleepStop`

INFO: WELD-ENV-002003: Weld SE container STATIC_INSTANCE initialized
## SleepBatchlet sleep.minutes: 5
Oct 10, 2022 12:37:50 PM org.jberet.runtime.JobExecutionImpl stop
INFO: JBERET000033: Stopping job execution: 31
in @Stop, Thread[main,5,main]
jobExecution id=31, batchStatus=STOPPED, exitStatus=STOPPED, jobParameters={sleep.minutes=6}, restartPosition=null, createTime=Mon Oct 10 12:37:48 EDT 2022, startTime=Mon Oct 10 12:37:49 EDT 2022, lastUpdateTime=Mon Oct 10 12:37:50 EDT 2022, endTime=Mon Oct 10 12:37:50 EDT 2022
`

**Comment by chschroe74** [comment]
You are completely right. My bad, sorry for that!
I initially encountered the problem in my Quarkus application using the Quarkus JBeret extension. I wanted to further narrow it down, so I tried to reproduce it with pure JBeret. When I saw the output, I thought that I had found it, and I completely overlooked that the different tests use different job definitions.
Feel free to close this ticket. I will dig deeper into the Quarkus extension to find out why the injection doesn't work. Again, my apologies for wasting your time.
